code/game/machinery/feeder.dm

```diff
@@ -0,0 +1,123 @@
+/obj/machinery/feeder
+	name = "\improper Feeder"
+	icon = 'icons/obj/feeder.dmi'
+	desc = "This is a feeder. Put in a regent container, then click and drag the feeder to someone!"
+	anchored = 0
+	density = 0
+
+
+/obj/machinery/feeder/var/mob/living/carbon/human/attached = null
+/obj/machinery/feeder/var/obj/item/weapon/reagent_containers/beaker = null
+
+/obj/machinery/feeder/update_icon()
+	if(attached)
+		icon_state = "feeding"
+	else
+		icon_state = ""
+
+	overlays = null
+
+	if(beaker)
+		var/datum/reagents/reagents = beaker.reagents
+		if(reagents.total_volume)
+			var/image/filling = image('icons/obj/feeder.dmi', src, "reagent")
+
+			var/percent = round((reagents.total_volume / beaker.volume) * 100)
+			switch(percent)
+				if(0 to 9)		filling.icon_state = "reagent0"
+				if(10 to 19) 	filling.icon_state = "reagent10"
+				if(20 to 44)	filling.icon_state = "reagent20"
+				if(45 to 59)	filling.icon_state = "reagent45"
+				if(60 to 74)	filling.icon_state = "reagent60"
+				if(75 to 89)	filling.icon_state = "reagent75"
+				if(90 to 94)	filling.icon_state = "reagent90"
+				if(95 to INFINITY)	filling.icon_state = "reagent100"
```

Comment:
I'm sure there's a better way to handle this, though I can't puzzle through it. If you'd like to attempt, check out L25 in `code\game\machinery\cell_charger.dm`